src/components/create-guild/Requirements/hooks/useCreateRequirement.ts

```diff
@@ -48,3 +49,51 @@ const useCreateRequirement = (
 }
 
 export default useCreateRequirement
+
+const useCreateRequirementForRole = () => {
```

Comment:
If this will be removed this is not relevant, but is there a reason for not using useSubmit here? That handles loading state and success / error actions we have here too